File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TiledMapTimeline.java
/**
 * 
 */
package com.abubusoft.xenon.animations;

import com.abubusoft.xenon.animations.events.EventFrameListener;

/**
 * @author Francesco Benincasa
 *
 */
public class TiledMapTimeline extends Timeline<TiledMapAnimation, TranslationFrame, TiledMapAnimationHandler> {

	public TiledMapTimeline()
	{
		handler=new TiledMapAnimationHandler();
	}
	
	/**
	 * Imposta il listener relativo allo spostamento.
	 * @param value
	 */
	public void setOnMoveEventListener(EventFrameListener<TranslationFrame> value) {
		handler.setFrameListener(value);
	}

	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/animations/TiledMapTimeline.kt
/**
 *
 */
package com.abubusoft.xenon.animations

import com.abubusoft.xenon.animations.events.EventFrameListener

/**
 * @author Francesco Benincasa
 */
class TiledMapTimeline : Timeline<TiledMapAnimation?, TranslationFrame?, TiledMapAnimationHandler?>() {
    init {
        handler = TiledMapAnimationHandler()
    }

    /**
     * Imposta il listener relativo allo spostamento.
     * @param value
     */
    fun setOnMoveEventListener(value: EventFrameListener<TranslationFrame?>?) {
        handler!!.setFrameListener(value)
    }
}